code/game/objects/items/devices/scanners.dm

```diff
@@ -38,56 +38,41 @@ REAGENT SCANNER
 	return ..()
 
 /obj/item/t_scanner/attack_self(mob/user)
-
 	on = !on
 	icon_state = copytext(icon_state, 1, length(icon_state))+"[on]"
-
 	if(on)
 		START_PROCESSING(SSobj, src)
-
+	else
+		STOP_PROCESSING(SSobj, src)
 
 /obj/item/t_scanner/process()
 	if(!on)
 		STOP_PROCESSING(SSobj, src)
 		return null
 	scan()
 
+// Show the holder the T-ray layout within `scan_range` for the `pulse_duration` deciseconds.
 /obj/item/t_scanner/proc/scan()
+	var/mob/viewer = src.loc
+	if(!ismob(viewer) || !viewer.client)
+		return
 
-	for(var/turf/T in range(scan_range, src.loc) )
-
-		if(!T.intact)
-			continue
-
+	var/list/t_ray_images = list()
+	for(var/turf/T in range(scan_range, viewer))
 		for(var/obj/O in T.contents)
 
 			if(O.level != 1)
 				continue
 
-			if(O.invisibility == 101)
-				O.invisibility = 0
-				O.alpha = 128
-				spawn(pulse_duration)
-					if(O)
-						var/turf/U = O.loc
-						if(U && U.intact)
-							O.invisibility = 101
-							O.alpha = 255
-		for(var/mob/living/M in T.contents)
-			var/oldalpha = M.alpha
-			if(M.alpha < 255 && istype(M))
-				M.alpha = 255
-				spawn(10)
-					if(M)
-						M.alpha = oldalpha
-
-		var/mob/living/M = locate() in T
-
-		if(M && M.invisibility == 2)
-			M.invisibility = 0
-			spawn(2)
-				if(M)
-					M.invisibility = INVISIBILITY_LEVEL_TWO
+			if(O.invisibility == INVISIBILITY_MAXIMUM || HAS_TRAIT(O, TRAIT_T_RAY_VISIBLE))

```

Comment:
This will make things appear that were not shown prior. Might want to take another look at what items will be visible due to this. As an example the poolcontroller will become visible now.

You might have to change those to have a more fitting invisibility value though